tests/test_async_client.py

```diff
@@ -0,0 +1,142 @@
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_validate_key_invalid_key_raises_InvalidKey_exception(self, mock_get):
+        mock_get.return_value = {'detail': 'Not authenticated'}, 403
+        with self.assertRaises(willofsteel.exceptions.InvalidKey):
+            await self.client._verify_key()
+
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_get_player_returns_player_object(self, mock_get):
+        mock_get.return_value = {'user_id': 1,
+                                 'registered_at': '2023-01-01 01:01:01'}, 200
+        response = await self.client.get_player()
+        self.assertIsInstance(response, Player)
+
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_get_player_inventory_returns_dictionary_of_items(self, mock_get):
+        mock_get.return_value = {
+            "items": {"BAKERY_TOKEN": 10, "HEALING_TOKEN": 1}}, 200
+        response = await self.client.get_player_inventory()
+        self.assertIsInstance(response, dict)
+        for item in response.values():
+            self.assertIsInstance(item, int)
+
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_get_player_army_returns_dictionary_of_units(self, mock_get):
+        mock_get.return_value = {
+            "units": {"infantry": 500, "cavalry": 100, "artillery": 50}}, 200
+        response = await self.client.get_player_army()
+        self.assertIsInstance(response, dict)
+        for unit in response.values():
+            self.assertIsInstance(unit, int)
+
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_get_outposts_returns_list_of_outpost_objects(self, mock_get):
+        mock_get.return_value = {'outposts': [{'profile_id': 'abc', 'name': 'an_outpost'}, {
+            'profile_id': 'xyz', 'name': 'another_outpost'}]}, 200
+        response = await self.client.get_outposts()
+        self.assertIsInstance(response, list)
+        for outpost in response:
+            self.assertIsInstance(outpost, Outpost)
+
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_get_alliance_in_alliance_returns_alliance_object(self, mock_get):
+        mock_get.return_value = {'owner': 1,
+                                 'created_at': '2023-01-01 01:-1:01',
+                                 'name': 'Alliance',
+                                 'user_limit': 50,
+                                 'bank': 1}, 200
+        response = await self.client.get_alliance()
+        self.assertIsInstance(response, Alliance)
+
+    @patch('willofsteel.async_client.AsyncClient.request', new_callable=AsyncMock)
+    async def test_get_alliance_not_in_alliance_raises_NotInAlliance_Exception(self, mock_get):
+        mock_get.return_value = {'detail': 'Not in an alliance'}, 400
```

Comment:
```suggestion
        mock_get.return_value = {"success": False, "detail": "alliance not found"}, 400
```

API does not return "Not in an alliance" as the detail, the function just checks for the status code, but the error messages should be correct.